tracer/src/Datadog.Trace/IAST/Propagation/StringModuleImpl.cs

```diff
@@ -136,6 +138,133 @@ private static void OnStringSubSequence(string self, int beginIndex, object resu
         }
     }
 
+    public static string OnStringJoin(string result, IEnumerable<object> values, int startIndex = 0, int count = -1)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(result))
+            {
+                return result;
+            }
+
+            var iastContext = IastModule.GetIastContext();
+            if (iastContext == null)
+            {
+                return result;
+            }
+
+            TaintedObjects taintedObjects = iastContext.GetTaintedObjects();
+
+            var newRanges = new List<Range>();
+            int pos = 0;
+            int i = 0;
+            foreach (var element in values)
+            {
+                if (i >= startIndex && (count < 0 || i < startIndex + count))
+                {
+                    pos = GetPositionAndUpdateRangesInStringJoin(taintedObjects, newRanges, pos, null, 1, element?.ToString() ?? string.Empty, false);
+                }
+
+                i++;
+            }
+
+            if (newRanges.Count > 0)
+            {
+                taintedObjects.Taint(result, newRanges.ToArray());
```

Comment:
Not related to this PR, but I wonder if we could avoid the array creation here.

Reply:
We need a new array with the new ranges for the new tainted object. It is expensive and I have thought about it, but I do not see any other solution.